spec/models/user_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it { should have_valid(:first_name).when('John', 'Sally') }
+  it { should_not have_valid(:first_name).when(nil, '') }
+
+  it { should have_valid(:last_name).when('Smith', 'Adams') }
+  it { should_not have_valid(:last_name).when(nil, '') }
+
+  it { should have_valid(:email).when('user@example.com', 'bill@gmail.com') }
+  it { should_not have_valid(:email).when(nil,
```

Comment:
Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.